### ISSUE TITLE
Auto PGD doesn't not support Kerasclassifier wrapper? 

### ISSUE THREAD

**Issue by sfarkya** [Trusted-AI/adversarial-robustness-toolbox#643]
I have been trying to apply Auto PGD attack on a KerasClassifier wrapped model and I was getting the error shown in the screenshot below: 

I simplified my issue using the **get_image_classifier_kr** model available in the test case of the repo. 

![Screen Shot 2020-10-06 at 10 55 32 PM](https://user-images.githubusercontent.com/6245078/95282341-3328f100-0827-11eb-9e12-febbd04d6102.png)

Upon further looking into the code for Auto PGD it seems that the attack doesn't support KerasClassifier. Is there any specific reason for that? Or Should put this as a feature request? 

[https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/5524d20788c62fbf405e6af9f81243c1e933ea94/art/attacks/evasion/auto_projected_gradient_descent.py#L87](https://github.com/Trusted-AI/adversarial-robustness-toolbox/blob/5524d20788c62fbf405e6af9f81243c1e933ea94/art/attacks/evasion/auto_projected_gradient_descent.py#L87)  

Thank you!

**Comment by beat-buesser** [comment]
Hi @sfarkya Thank you for using ART!

Yes, that's correct, `AutoProjectedGradientDescent` in ART 1.4 supports only `TensorFlowClassifier`, `TensorFlowV2Classifier`, and `PyTorchClassifier`. We might add support for all classifiers and user-defined loss functions to the upcoming ART 1.5.

**Comment by sfarkya** [comment]
Thanks! @beat-buesser 
I will look forward to it! ART 1.4 was just released so it would take a while releasing the 1.5.

**Comment by beat-buesser** [comment]
Hi @sfarkya The current plan is to release ART 1.5 on Nov 30, the updates might become available earlier on branch `dev_1.5.0`. If  you are interested you could start to extend `AutoProjectedGradientDescent` to accept classifiers without modifying them as an option and maybe make a pull request? I would be happy to provide support.

**Comment by beat-buesser** [comment]
Hi @sfarkya I have implemented some updates to `AutoProjectedGradientDescent` that make it compatible with `KerasClassifier`. The updates are currently on branch `development_apgd`. If you have a chance to try them out please let us know if it works for you.

**Comment by aldahdooh** [comment]
Hi @beat-buesser 

The support of ``KerasClassifier`` is not yet in ART1.5.1?
